qtp_biom/summary.py

```diff
@@ -29,8 +30,14 @@
 
 def _generate_html_summary(biom_fp, metadata, out_dir, is_analysis, tree=None):
     if is_analysis:
-        metadata = qiime2.Metadata(pd.DataFrame.from_dict(
-            metadata, orient='index'))
+        # we need to save and load the df so qiime does it's magic for parsing
+        # columns
+        fd, path = mkstemp()
+        df = pd.DataFrame.from_dict(metadata, orient='index')
+        df.to_csv(path, index_label='#SampleID', na_rep='', sep='\t',
+                  encoding='utf-8')
+        metadata = qiime2.Metadata.load(path)
+        remove(path)
```

Comment:
This line should cause an error. path is a directory, and os.remove(directory) should raise an OSError. Also,  'fd, path = mkstemp()' potentially creates a new temp directory, but definitely creates a new temp file with an open file handle that is not being used. The open file handle may potentially cause conflicts when deleting the directory it's in.

Suggest using mkdtemp() instead of mkstemp(), which will create a temp directory in a secure manner. It will only return the path to a new temp directory, instead of the above tuple. Suggest using shutil.rmtree(path) to blow away the new temp directory and all files in it.